repos/system_upgrade/el7toel8/actors/sctpconfigread/actor.py

```diff
@@ -0,0 +1,159 @@
+from leapp.actors import Actor
+from leapp.models import SCTPConfig
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+#
+# Generic functions
+#
+def check_loaded_module(module):
+    """Check if a given module is loaded."""
+    import os
+    try:
+        os.stat('/sys/module/%s' % module)
+    except OSError:
+        logging.debug('Module %s is not loaded.' % module)
+        return False
+    logging.debug('Module %s is loaded.' % module)
+    return True
+
+def check_any_file_present(files):
+    import os
+    for f in files:
+        try:
+            if os.stat(f):
+                logging.debug('File %s is present.' % module)
+                return True
+        except OSError:
+            continue
+    return False
+
+#
+# Functions for probing for SCTP usage
+#
+def check_dlm_cfgfile():
+    """Parse the config file"""
+    fname = "test.conf"
+    try:
+        fp = open(fname, 'r')
+    except:
+        return False
+
+    cfgs = '[dlm]\n' + fp.read()
+    del fp
+
+    try:
+        from configparser import SafeConfigParser
+        cfg = SafeConfigParser()
+        cfg.read_string(fp)
+    except:
+        # Python2 ConfigParser doesn't have cfg.read_string
+        from cStringIO import StringIO
+        from ConfigParser import SafeConfigParser
+        cfg = SafeConfigParser()
+        cfg.readfp(StringIO(cfgs))
+
+    proto = cfg.get('dlm', 'protocol').lower()
+    if proto in ['sctp', 'detect', '1', '2']:
+        return True
+
+    return False
+
+def check_dlm_sysconfig():
+    """Parse /etc/sysconfig/dlm"""
+    try:
+        #fp = open('/etc/sysconfig/dlm', 'r')
+        fp = open('dlm', 'r')
+    except:
+        return False
+
+    import re
+    regex = re.compile('^[^#]*DLM_CONTROLD_OPTS.*=.*(?:--protocol|-r)[ =]*([^ ]+).*', re.IGNORECASE)
+    for line in fp.readlines():
+        if regex.match(line):
+            proto = regex.sub('\\1', line).lower().strip()
+            if proto in ['sctp', 'detect']:
+                return True
+
+def is_dlm_using_sctp():
+    if check_dlm_cfgfile():
+        logging.info('DLM is configured to use SCTP on dlm.conf.')
+        return True
+
+    if check_dlm_sysconfig():
+        logging.info('DLM is configured to use SCTP on sysconfig.')
+        return True
+
+    return False
+
+def is_sctp_used():
+    # If anything is using SCTP, be it for listening on new connections or
+    # connecting somewhere else, the module will be loaded. Thus, no need to
+    # also probe on sockets.
+    if check_loaded_module('sctp'):
+        return True
+
+    # Basic files from lksctp-tools. This check is enough and checking RPM database
+    # is an overkill here.
```

Comment:
You can also consume the `leapp.models.InstalledRPM` Model and check against them if the lksctp-tools are installed. But this should be ok, if this is safe enough for your test.